File: 프로그래머스/1/12926. 시저 암호/시저 암호.swift
func solution(_ s:String, _ n:Int) -> String {
    var result = ""
    var asc_arr:[Int] = s.map { Int(UnicodeScalar(String($0))!.value) }
    for a in asc_arr { 
        if (a == Int(UnicodeScalar(" ").value)) { result += " " }
        else {     
            var asc = a+n
            if (a >= 65 && a <= 90) {
                if (a+n > 90) { asc -= 26 }
            } else {
                if (a+n > 122) { asc -= 26 }
            }
            result += String(UnicodeScalar(asc)!) 
        }
    }
    return result
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";
        String[] list = s.split("");
        for(int i=0; i<s.length(); i++) {
            if (s.charAt(i) == ' ') answer += " ";
            else {
                int asc = (int) s.charAt(i);
                asc += n;
                if (!Character.isAlphabetic((char)asc) 
                    || (Character.isUpperCase(s.charAt(i)) 
                        && Character.isLowerCase((char)asc))
                   ) asc -= 26 ;
                answer += Character.toString((char)asc);
            }
        }
        return answer;
    }
}